src/containers/editEvent/index.tsx

```diff
@@ -0,0 +1,149 @@
+  asyncRequestIsComplete,
+  asyncRequestIsFailed,
+  asyncRequestIsLoading,
+  asyncRequestIsNotStarted,
+  AsyncRequestLoading,
+  AsyncRequestNotStarted,
+} from '../../utils/asyncRequest';
+import EventsForm, {
+  EventsFormInitialValues,
+} from '../../components/EventsForm';
+import FormInitialText from '../../components/FormInitialText';
+import { Alert, Spin, Typography } from 'antd';
+import { editEvent } from '../createEvent/ducks/thunks';
+import { EventsFormData } from '../../components/EventsForm';
+import { EventInformation } from '../upcoming-events/ducks/types';
+import protectedApiClient from '../../api/protectedApiClient';
+import moment from 'moment';
+import { encodeProfileFieldFile } from '../../utils/fileEncoding';
+
+interface EditEventProps {
+  readonly tokens: UserAuthenticationReducerState['tokens'];
+}
+
+const { Title } = Typography;
+
+interface SingleEventParams {
+  id: string;
+}
+
+const EditEventContainer: React.FC<EditEventProps> = ({ tokens }) => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const id = Number(useParams<SingleEventParams>().id);
+  const [editEventRequest, setEditEventRequest] = useState<
+    AsyncRequest<void, any>
+  >(AsyncRequestNotStarted());
+
+  const [event, setEvent] = useState<AsyncRequest<EventInformation, any>>(
+    AsyncRequestNotStarted(),
+  );
+
+  const onFinish = async (data: EventsFormData) => {
+    const eventPicture =
+      data.thumbnail && (await encodeProfileFieldFile(data.thumbnail));
+
+    try {
+      setEditEventRequest(AsyncRequestLoading());
+      await dispatch(
+        editEvent(id, {
+          title: data.title,
+          capacity: data.capacity,
+          thumbnail: eventPicture,
+          price: data.price,
+          details: {
+            description: data.description,
+            location: data.location,
+            start: data.start,
+            end: data.end,
+          },
+        }),
+      );
+      setEditEventRequest(AsyncRequestCompleted(undefined));
+    } catch (err) {
+      setEditEventRequest(AsyncRequestFailed(err));
+    }
+  };
+  const mapEventInfoToFormData = (
+    info: EventInformation,
+  ): EventsFormInitialValues => {
+    return {
+      title: info.title,
+      capacity: info.capacity,
+      thumbnail: info.thumbnail,
+      price: info.price,
+      description: info.details.description,
+      location: info.details.location,
+      start: moment(info.details.start),
+      end: moment(info.details.end),
+    };
+  };
+
+  useEffect(() => {
+    if (asyncRequestIsNotStarted(event)) {
+      setEvent(AsyncRequestLoading());
+      protectedApiClient
+        .getEventInfoById(id)
+        .then((res) => {
+          setEvent(AsyncRequestCompleted(res));
+        })
+        .catch((error: any) => {
+          setEvent(AsyncRequestFailed(error));
+        });
+    }
+  }, [event, id]);
+
+  if (asyncRequestIsComplete(editEventRequest)) {
+    history.push('/events/' + id);
```

Comment:
If you really wanna be consistent, you could use the base constant you defined in the delete file